File: PythonIntro/Python/04_01SimpleClasses.py
from abc import ABCMeta
from abc import abstractmethod

# A SIMPLE CLASS

class Counter(object):
    def __init__(self):
        self._count = 0

    def increment(self):
        self._count += 1

    def decrement(self):
        self._count -=1

    #def __str__(self):
    #    return str(self._count)


if __name__ == "__main__":
    counter1 = Counter()
    counter2 = Counter()

    counter1.increment()
    counter1.increment()
    counter1.decrement()
    counter2.decrement()
    counter2.decrement()
    counter2.decrement()
    counter1.increment()

    print("counter1: ", counter1)
    print("counter2: ", counter2)

File: PythonIntro/CSharp/04_01SimpleClasses.cs
using System;
using System.Globalization;

namespace CSharp
{
    public class Counter
    {
        int count;

        public Counter()
        {
            count = 0;
        }

        public void Increment()
        {
            count++;
        }

        public void Decrement()
        {
            count--;
        }

        public override string ToString()
        {
            return count.ToString(CultureInfo.CurrentUICulture);
        }

        public void Run()
        {
            var counter1 = new Counter();
            var counter2 = new Counter();

            counter1.Increment();
            counter1.Increment();
            counter1.Decrement();
            counter2.Decrement();
            counter2.Decrement();
            counter2.Decrement();
            counter1.Increment();

            Console.WriteLine("counter1: {0}", counter1);
            Console.WriteLine("counter2: {0}", counter2);
        }
    }
}
